File: skill_sdk/tests/fixtures/ts-project/src/shop/payment/StripeGateway.ts
import { PaymentGateway } from './PaymentGateway';
import { PaymentResult } from './PaymentResult';

/**
 * Stripe-backed payment gateway.
 */
export class StripeGateway implements PaymentGateway {
  constructor(private readonly apiKey: string) {}

  charge(amount: number, currency: string): PaymentResult {
    const txId = `stripe_ch_${this.hashFloat(amount)}`;
    return new PaymentResult(true, txId, `Charged ${amount} ${currency}`);
  }

  refund(transactionId: string): PaymentResult {
    return new PaymentResult(
      true,
      `stripe_ref_${transactionId}`,
      `Refunded transaction ${transactionId}`
    );
  }

  getApiKey(): string {
    return this.apiKey;
  }

  private hashFloat(f: number): number {
    return Math.floor(f * 100);
  }
}

File: skill_sdk/tests/fixtures/cpp-project/src/shop/StripeGateway.cpp
#include "StripeGateway.h"

StripeGateway::StripeGateway(const std::string& apiKey) : apiKey_(apiKey) {}

PaymentResult StripeGateway::Charge(double amount, const std::string& currency) {
    return PaymentResult{
        true,
        "stripe_ch_" + std::to_string(static_cast<int>(amount * 100)),
        "Charged " + std::to_string(amount) + " " + currency
    };
}

PaymentResult StripeGateway::Refund(const std::string& transactionID) {
    return PaymentResult{
        true,
        "stripe_ref_" + transactionID,
        "Refunded transaction " + transactionID
    };
}
